web/packages/teleport/src/Assist/contexts/messages.tsx

```diff
@@ -351,30 +358,74 @@ export async function setConversationTitle(
   });
 }
 
+const TEN_MINUTES = 10 * 60 * 1000;
+
 const logger = Logger.create('assist');
 
 export function MessagesContextProvider(
   props: PropsWithChildren<MessagesContextProviderProps>
 ) {
-  const { conversationId } = useParams<{ conversationId: string }>();
   const { clusterId } = useStickyClusterId();
 
   const [error, setError] = useState<string>(null);
   const [loading, setLoading] = useState(true);
   const [responding, setResponding] = useState(false);
   const [messages, setMessages] = useState<Message[]>([]);
 
-  const socketUrl = cfg.getAssistConversationWebSocketUrl(
-    getHostName(),
-    clusterId,
-    getAccessToken(),
-    props.conversationId
-  );
+  const webSocketRef = useRef<WebSocket>(null);
 
-  const { sendMessage, lastMessage } = useWebSocket(socketUrl, {
-    share: true, // when share is false the websocket tends to disconnect
-    retryOnError: true,
-  });
+  const setupWebSocket = useCallback(() => {
+    if (webSocketRef.current) {
+      webSocketRef.current.close();
+    }
+
+    const socketUrl = cfg.getAssistConversationWebSocketUrl(
+      getHostName(),
+      clusterId,
+      getAccessToken(),
+      props.conversationId
+    );
+
+    webSocketRef.current = new WebSocket(socketUrl);
+
+    webSocketRef.current.onmessage = async (event: MessageEvent) => {
+      const value = JSON.parse(event.data) as ServerMessage;
+
+      // When a streaming message ends, or a non-streaming message arrives
+      if (
+        value.type === 'CHAT_PARTIAL_MESSAGE_ASSISTANT_FINALIZE' ||
+        value.type === 'COMMAND' ||
+        value.type === 'CHAT_MESSAGE_ASSISTANT' ||
+        value.type === 'CHAT_MESSAGE_ERROR'
+      ) {
+        setResponding(false);
+      }
+
+      convertServerMessage(value, clusterId).then(res => {
+        setMessages(prev => {
+          const curr = [...prev];
+          res(curr);
+          return curr;
+        });
+      });
+    };
+  }, []);
+
+  useEffect(() => {
+    setupWebSocket();
+
+    // refresh the websocket connection every 10 minutes to avoid
+    // session timeouts
+    const id = window.setInterval(setupWebSocket, TEN_MINUTES);
```

Comment:
This should be less than the deadline
```suggestion
    const id = window.setInterval(setupWebSocket, TEN_MINUTES * 0.8);
```